### PR TITLE
Fix navigation block classname issues

### PR DESCRIPTION
**Pull request by jasmussen**
## Description

Followup to issues caused by #34171. Classnames were changed and made generic, this PR catches a few that I missed the first time around.

## How has this been tested?

Please test the navigation block empty, empty unselected, with menu items including dropdowns, with the Page List that also includes submenu items. 

Please test the navigation scren.

Note that while this PR slightly improves the navigation screen, it doesn't go as far as #34281 does, which includes some further CSS changes to make the chevron metrics and rotation work better. Happy to include this here if need be, but wanted to keep it to just classname fixes for a start.

## Screenshots 

Before:

Screenshot 2021-08-27 at 08 45 49

After:

Screenshot 2021-08-27 at 09 33 43

Screenshot 2021-08-27 at 09 33 48

Screenshot 2021-08-27 at 09 37 19
Screenshot 2021-08-27 at 09 38 07

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/components/src/navigation/README.md [hunk 1]
```diff
@@ -44,6 +44,15 @@ const MyNavigation = () => (
 	</Navigation>
 );
 ```
+## CSS Classes leveraged
```
Comment (jasmussen) [inline packages/components/src/navigation/README.md:47]: Let me know if this is the right place and/or format for these.

Comment (talldan) [reply packages/components/src/navigation/README.md:47]: Looks good, thanks for doing that. 👍

**Comment by talldan** [comment]
@jasmussen I'm seeing a wider range of styling issues for the core navigation block editor styles. For this I'm turning off the 'Use theme styles' option in the editor to see what the base styles are like:
- Add a horizontal nav block, click start empty. The block appender when for an empty nav block appears in bottom right corner of the block, while I would expect it to be on the left.
Screenshot 2021-08-27 at 4 15 17 pm

- Add a page link to the empty nav block, the page link block ends up about 1/3rd across the page from other content. I'd again expect it to be aligned to the left
Screenshot 2021-08-27 at 4 16 19 pm

- Deselect the block, page link jumps to the right side of the screen
Screenshot 2021-08-27 at 4 17 45 pm

- Add a second page link and add a submenu, the submenu jumps to the right side of the screen with the submenu partially offscreen
Screenshot 2021-08-27 at 4 18 30 pm

- Try using the navigation block's Justify Items options, they have no effect.

- Add a new vertical navigation block and a couple of page links. There doesn't seem to be any spacing between the links:
Screenshot 2021-08-27 at 4 20 22 pm

- Type in one of the page links in the vertical navigation block, the appender moves right.

**Comment by jasmussen** [comment]
Thank you for the thorough testing! I'll get those fixed up 👌

**Comment by jasmussen** [comment]
Thanks again for the excellent bug report, it helped me trace the issue to #34258. I believe the change I just pushed fixes all the issues you mention:

fix

**Comment by jasmussen** [comment]
> Add a new vertical navigation block and a couple of page links. There doesn't seem to be any spacing between the links:

I wanted to expand on this one, this lack of margin is intentional so as to enable navigation designs that are meant to be "lists of links". Some more details in #30805. You can still adjust the line height, or add background color, which both affect the vertical spacing:

Screenshot 2021-08-27 at 10 34 18

Screenshot 2021-08-27 at 10 33 24

Those margins also have reduced specificity so that you can set them via global styles.

**Review by talldan — APPROVED** [review]
Thanks, lets move forwards with this. I'm still seeing many of the issues I mentioned (testing using TT1 with theme editor styles disabled). Hopefully they can be fixed as a followup.

**Comment by jasmussen** [comment]
Thanks for the review. There are [some issues that have to be fixed in TT1 itself](https://core.trac.wordpress.org/ticket/53220), and which we can't fix in the navigation block lest we break global styles.

**Pull request merged**